src/Microsoft.Extensions.ObjectPool/ObjectPool.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.ObjectPool
+{
+    public abstract class ObjectPool<T> where T : class
```

Comment:
I wanted to wait and see if we any use cases for `shared`. I don't want to just parrot what we were pitching the BCL team because we don't have the same requirements.

For instance, we should probably have the provider, but not `ObjectPoolProvider.Current` or `ObjectPool<T>.Shared`. Then we can put the provider in DI, and allow replacement for things like leak detection.